### PR TITLE
MGDAPI-4906 restore workshop type cloud resource config

### PR DESCRIPTION
**Pull request by cecobask**
# Issue link
https://issues.redhat.com/browse/MGDAPI-4906

# What
Add in the workshop type to the cloud-resource-config.
Used for creating on cluster Redis and Postgres for test cases.

# Verification steps
## Local
- Prepare your cluster, run RHOAM locally and await installation complete:
  ```
  LOCAL=false make cluster/prepare/local
  LOCAL=false make deploy/integreatly-rhmi-cr.yml
  LOCAL=false make code/run
  ```
- Create a `workshop` Postgres instance and await its completion:
  ```
  cat << EOF | oc create -f - -n redhat-rhoam-operator                   
    apiVersion: integreatly.org/v1alpha1
    kind: Postgres
    metadata:
      name: throw-away-postgres
      labels:
        productName: productName
    spec:
      secretRef:
        name: throw-away-postgres-sec
      tier: development
      type: workshop
  EOF
  ```
- Proceed with cleanup and await its completion:
  ```
  oc delete postgres throw-away-postgres -n redhat-rhoam-operator
  oc delete rhmi rhoam -n redhat-rhoam-operator
  LOCAL=false make cluster/cleanup
  ```

## OLM
- Install RHOAM through OLM using an index pointing at the latest GitHub tag and await its completion:
  ```
  git switch -d rhoam-v1.29.0
  LOCAL=false make cluster/prepare/local
  LOCAL=false make deploy/integreatly-rhmi-cr.yml
  cat << EOF | oc create -f - -n openshift-marketplace
    apiVersion: operators.coreos.com/v1alpha1
    kind: CatalogSource
    metadata:
      name: rhoam-operators
      namespace: openshift-marketplace
    spec:
      image: 'quay.io/tdimov0/managed-api-service-index:1.29.0'
      sourceType: grpc
  EOF
  ```
- Upgrade RHOAM to a version containing my changes and await its completion:
  ```
  oc patch catalogsource rhoam-operators -n openshift-marketplace --type=json -p='[{"op": "replace", "path": /spec/image, "value": "quay.io/tdimov0/managed-api-service-index:1.30.0"}]'
  ```
- Create a `workshop` Postgres instance and await its completion:
  ```
  cat << EOF | oc create -f - -n redhat-rhoam-operator                   
    apiVersion: integreatly.org/v1alpha1
    kind: Postgres
    metadata:
      name: throw-away-postgres
      labels:
        productName: productName
    spec:
      secretRef:
        name: throw-away-postgres-sec
      tier: development
      type: workshop
  EOF
  ```
- Proceed with cleanup and await its completion:
  ```
  oc delete postgres throw-away-postgres -n redhat-rhoam-operator
  oc delete rhmi rhoam -n redhat-rhoam-operator
  LOCAL=false make cluster/cleanup
  ```

### CODE REVIEW

**Review by Boomatang — COMMENTED** [review]
I am not seen why this is need. The Jira says it is for test but test were passing before hand. The new test added does not check that the configuration is valid but only if there is the correct phase response. 

If this workshop type is used for some kind of manual testing, it is not documented on how it should be used. 

The two questions I have are:
1. Why is this needed?
2. Is puting code for testing purposes into production code the correct way of address this need?

**Comment by cecobask** [comment]
@Boomatang If I understand correctly, this is needed for creating Postgres and Redis instances on cluster regardless of RHOAM install type. I see that if `useClusterStorage` is false, there currently isn't a way to create on cluster instances without the `workshop` type. Do you have a solution in mind that doesn't involve adding new config type?

**Comment by austincunningham** [comment]
@Boomatang there are definitely some manual test cases that require on cluster postgres and redis as that is what lead me to discover that it had been removed. The workshop type is the easiest way to create on cluster postgres/redis. I believe that there may be some sre SOP's that also require this also. And in general we have used the workshop type to create on cluster postgres during trouble shooting.

**Comment by Boomatang** [comment]
The naming of this configuration is misleading and did align with an rhmi install type of workshop. Is the `workshop` type a know type in cro or can it be renamed to something more meaningful. This feature needs to be documented, I for one, had no idea this was possible and if ever need a redis pod, I would have deploy one manually.

File: docs/tests/workshop.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# workshop
```
Comment (Boomatang) [inline docs/tests/workshop.md:1]: This title should really match the title in the navigation 
```markdown
# Workshop postgres/redis instances
```

File: docs/tests/workshop.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+The workshop type is being used for creating on cluster, throwaway postgres/redis instances for troubleshooting. 
+While this isn't functionality used in production, there are some manual test cases and sops that rely on it.
+
+## Usage
+To create a workshop instance, log in to a cluster and issue one of the following commands:
+- Postgres
```
Comment (Boomatang) [inline docs/tests/workshop.md:7]: The lack of extra line above is causing this bullet point not to be rendered by mkdocs. I also think this reads better if it a heading and not a bullet point.
```markdown
### Postgres
```

File: docs/tests/workshop.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    labels:
+      productName: productName
+  spec:
+    secretRef:
+      name: throw-away-postgres-sec
+    tier: development
+    type: workshop
+EOF
+```
+- Redis
```
Comment (Boomatang) [inline docs/tests/workshop.md:23]: If the postgres section is being turned to a heading then this section should also be turned to a heading.
```markdown
### Redis
```

**Review by Boomatang — CHANGES_REQUESTED** [review]
Functionaly, this is working as expected.

Looking for a few small changes in the docs section.

**Pull request merged**